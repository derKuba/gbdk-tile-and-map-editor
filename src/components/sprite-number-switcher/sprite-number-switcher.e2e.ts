import { newE2EPage } from '@stencil/core/testing';

describe('mvjp-sprite-number-switcher', () => {
    it('renders', async () => {
        const page = await newE2EPage();
        await page.setContent('<mvjp-sprite-number-switcher></mvjp-sprite-number-switcher>');

        const element = await page.find('mvjp-sprite-number-switcher');
        expect(element).toHaveClass('hydrated');
    });
});
