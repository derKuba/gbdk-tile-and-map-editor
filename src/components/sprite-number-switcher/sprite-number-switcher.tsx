import { Component, Host, h, Prop } from '@stencil/core';
import { Settings } from '../../models/settingsModel';
import Tunnel from '../../provider/StateProvider';

@Component({
    tag: 'mvjp-sprite-number-switcher',
    styleUrl: 'sprite-number-switcher.css',
    shadow: true,
})
export class SpriteNumberSwitcher {
    /**
     *   global matrix setter
     */
    @Prop() changeTileMatrix: Function;

    /**
     *   which sprite tile is active
     */
    @Prop() indexTileMatrix: number;

    /**
     *   set global sprite tile
     */
    @Prop() updateSprite: Function;

    /**
     *   global settings
     */
    @Prop() settings: Settings;

    private onClick = index => {
        this.changeTileMatrix(index);
    };

    render() {
        const { spriteAmount } = this.settings;

        const tiles = new Array(spriteAmount)
            .fill(0, 0, spriteAmount)
            .map((_, i) => (
                <mvjp-sprite-number-switcher-tile
                    index={i}
                    isActive={this.indexTileMatrix === i}
                    clickTile={this.onClick}
                    key={`index_${i}`}
                />
            ));

        return <Host>{tiles}</Host>;
    }
}
Tunnel.injectProps(SpriteNumberSwitcher, ['settings', 'changeTileMatrix', 'indexTileMatrix']); // eslint-disable-line @stencil/ban-side-effects
