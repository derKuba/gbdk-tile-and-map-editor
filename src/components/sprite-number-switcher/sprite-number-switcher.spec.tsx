import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { SpriteNumberSwitcher } from './sprite-number-switcher';

describe('mvjp-sprite-number-switcher', () => {
    it('builds', () => {
        expect(new SpriteNumberSwitcher()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const settings = {
            fieldWidth: 8,
            fieldHeight: 8,
            spriteAmount: 12,
            isGridVisible: true,
            isFieldNumberVisible: true,
        };

        const page = await newSpecPage({
            components: [SpriteNumberSwitcher],
            template: () => {
                return (
                    <mvjp-sprite-number-switcher
                        changeTileMatrix={() => {}}
                        choosenSprite={0}
                        settings={settings}
                        updateSprite={() => {}}></mvjp-sprite-number-switcher>
                );
            },
        });

        expect(page.root).toMatchSnapshot();
    });
});
