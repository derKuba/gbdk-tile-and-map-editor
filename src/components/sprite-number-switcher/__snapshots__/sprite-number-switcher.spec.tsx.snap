// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-sprite-number-switcher matches snapshot 1`] = `
<mvjp-sprite-number-switcher choosensprite="0">
  <mock:shadow-root>
    <mvjp-sprite-number-switcher-tile index="0"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="1"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="2"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="3"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="4"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="5"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="6"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="7"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="8"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="9"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="10"></mvjp-sprite-number-switcher-tile>
    <mvjp-sprite-number-switcher-tile index="11"></mvjp-sprite-number-switcher-tile>
  </mock:shadow-root>
</mvjp-sprite-number-switcher>
`;
